src/frontend/planner_test/tests/testdata/bushy_join.yaml

```diff
@@ -609,30 +609,31 @@
-                  ├─StreamExchange { dist: HashShard(orders.o_orderkey) }
-                  | └─StreamTableScan { table: orders, columns: [orders.o_orderkey, orders.o_orderdate], pk: [orders.o_orderkey], dist: UpstreamHashShard(orders.o_orderkey) }
-                  └─StreamExchange { dist: HashShard(lineitem.l_orderkey) }
-                    └─StreamHashJoin { type: Inner, predicate: partsupp.ps_suppkey = lineitem.l_suppkey AND partsupp.ps_partkey = lineitem.l_partkey, output: [partsupp.ps_supplycost, lineitem.l_orderkey, lineitem.l_suppkey, lineitem.l_quantity, lineitem.l_extendedprice, lineitem.l_discount, partsupp.ps_partkey, partsupp.ps_suppkey, part.p_partkey, lineitem.l_linenumber] }
-                      ├─StreamExchange { dist: HashShard(partsupp.ps_partkey, partsupp.ps_suppkey) }
-                      | └─StreamTableScan { table: partsupp, columns: [partsupp.ps_partkey, partsupp.ps_suppkey, partsupp.ps_supplycost], pk: [partsupp.ps_partkey, partsupp.ps_suppkey], dist: UpstreamHashShard(partsupp.ps_partkey, partsupp.ps_suppkey) }
-                      └─StreamExchange { dist: HashShard(lineitem.l_partkey, lineitem.l_suppkey) }
-                        └─StreamHashJoin { type: Inner, predicate: part.p_partkey = lineitem.l_partkey, output: [lineitem.l_orderkey, lineitem.l_partkey, lineitem.l_suppkey, lineitem.l_quantity, lineitem.l_extendedprice, lineitem.l_discount, part.p_partkey, lineitem.l_linenumber] }
-                          ├─StreamExchange { dist: HashShard(part.p_partkey) }
-                          | └─StreamProject { exprs: [part.p_partkey] }
-                          |   └─StreamFilter { predicate: Like(part.p_name, '%yellow%':Varchar) }
-                          |     └─StreamTableScan { table: part, columns: [part.p_partkey, part.p_name], pk: [part.p_partkey], dist: UpstreamHashShard(part.p_partkey) }
-                          └─StreamExchange { dist: HashShard(lineitem.l_partkey) }
-                            └─StreamTableScan { table: lineitem, columns: [lineitem.l_orderkey, lineitem.l_partkey, lineitem.l_suppkey, lineitem.l_quantity, lineitem.l_extendedprice, lineitem.l_discount, lineitem.l_linenumber], pk: [lineitem.l_orderkey, lineitem.l_linenumber], dist: UpstreamHashShard(lineitem.l_orderkey, lineitem.l_linenumber) }
+          └─StreamProject { exprs: [nation.n_name, Extract('YEAR':Varchar, orders.o_orderdate) as $expr1, ((lineitem.l_extendedprice * (1:Int32 - lineitem.l_discount)) - (partsupp.ps_supplycost * lineitem.l_quantity)) as $expr2, partsupp.ps_partkey, partsupp.ps_suppkey, nation.n_nationkey, supplier.s_suppkey, part.p_partkey, orders.o_orderkey, lineitem.l_orderkey, lineitem.l_linenumber] }
```

Comment:
The TPCH q9 presents a slight deviation from our anticipated join tree, despite both having an equal join height of 3. It appears that the current algorithm selects `merge_node` from `for merge_node in &n.relations {` without any priority consideration. In my opinion, we should prioritize selecting `merge_node` with lower join tree heights and fewer adjacent nodes.